cpp/src/parquet/arrow/schema_internal.cc

```diff
@@ -33,7 +33,10 @@ using ::arrow::internal::checked_cast;
 
 Result<std::shared_ptr<ArrowType>> MakeArrowDecimal(const LogicalType& logical_type) {
   const auto& decimal = checked_cast<const DecimalLogicalType&>(logical_type);
-  return ::arrow::Decimal128Type::Make(decimal.precision(), decimal.scale());
+  if (decimal.precision() <= ::arrow::Decimal128Type::kMaxPrecision) {
+    return ::arrow::Decimal128Type::Make(decimal.precision(), decimal.scale());
+  }
+  return ::arrow::Decimal256Type::Make(decimal.precision(), decimal.scale());
```

Comment:
While we are at it, can we catch cases where the precision doesn't fit in `Decimal256` and error out?

Reply:
This is done in the [Make](https://github.com/apache/arrow/blob/master/cpp/src/arrow/type.cc#L801) functions (same as lower bound for Decimal128